client/CPlayerInterface.h

```diff
@@ -56,11 +56,9 @@ namespace boost
 }
 
 /// Central class for managing user interface logic
-class CPlayerInterface : public CGameInterface, public IUpdateable
+class CPlayerInterface : public CGameInterface, public IUpdateable, public ArtifactsUIController
```

Comment:
Composition is generally better than inheritance.

Personally I would have used `std::unique_ptr<ArtifactsUIController> artifactController` member instead.